apps/condo/domains/resident/schema/DiscoverServiceConsumersService.test.js

```diff
@@ -3,275 +3,1096 @@
+            const unitType2 = FLAT_UNIT_TYPE
+            const unitName2 = faker.lorem.word()
+
+            const [billingAccount1] = await createTestBillingAccount(admin, billingIntegrationContext1, billingProperty1, {
+                unitType: unitType1,
+                unitName: unitName1,
             })
-            expect(createdServiceConsumersTotal).toBe(1)
-            expect(createdServiceConsumers[0].organization.id).toEqual(user.organization.id)
-            expect(createdServiceConsumers[0].accountNumber).toEqual(billingAccount.number)
-        })
 
-        test('discover multiple service consumers for specific resident', async () => {
-            const { context } = await makeContextWithOrganizationAndIntegrationAsAdmin(
-                {}, {}, { status: 'Finished' }
-            )
+            const [billingAccount2] = await createTestBillingAccount(admin, billingIntegrationContext2, billingProperty2, {
+                unitType: unitType2,
+                unitName: unitName2,
+            })
 
-            const [billingProperty] = await createTestBillingProperty(admin, context, { address: user.property.address })
-            const [resident] = await createTestResident(admin, user.user, user.property,
-                { address: billingProperty.address }
-            )
-            const [billingAccount1] = await createTestBillingAccount(admin, context, billingProperty,
-                { unitName: resident.unitName, unitType: resident.unitType }
-            )
+            const [billingAccount2a] = await createTestBillingAccount(admin, billingIntegrationContext2a, billingProperty2a, {
+                unitType: unitType2,
+                unitName: unitName2,
+            })
 
-            const [billingAccount2] = await createTestBillingAccount(admin, context, billingProperty,
-                { unitName: resident.unitName, unitType: resident.unitType }
-            )
+            // no consumers should be created after billing accounts appeared in database
+            // because of there are no residents yet
+            await waitFor(async () => {
+                const createdServiceConsumers = await ServiceConsumer.getAll(admin, {
+                    OR: [
+                        { AND: [{ organization: { id: user1.organization.id } }, { accountNumber: billingAccount1.number }] },
+                        { AND: [{ organization: { id: user2.organization.id } }, { accountNumber: billingAccount2.number }] },
+                        { AND: [{ organization: { id: user2a.organization.id } }, { accountNumber: billingAccount2a.number }] },
+                    ],
+                    deletedAt: null,
+                })
+
+                expect(createdServiceConsumers).toHaveLength(0)
+            }, { delay: 500 })
+
+            // Now add residents and start the mutation manually, to pass all billing accounts
+            const [resident1] = await createTestResident(admin, user1.user, user1.property, {
```

Comment:
user was acquired from `makeClientWithProperty`, so it must be staff by default.

Reply:
You should create new users for residents

Reply:
Done